docs/development/virtual_environments.rst

```diff
@@ -315,3 +328,116 @@ By default, direct SSH access is not enabled in the prod environment. You will n
 in over Tor after initial provisioning or set ``enable_ssh_over_tor`` to "false"
 during ``./securedrop-admin tailsconfig``. See :ref:`ssh_over_tor` or :ref:`ssh_over_local`
 for more info.
+
+.. _molecule_upgrade:
+
+Molecule: upgrade testing
+-------------------------
```

Comment:
The entire "Upgrade testing" section, starting here and running to the end of the document, would fit better as its own topic under the Developer Documentation heading. So:

1. Break out this section to its own doc, e.g. `docs/development/upgrade_testing.rst`
2. Update `docs/index.rst` to include the new document in the nav
3. Revise header levels (if necessary; haven't tested locally)
4. Revise title to be "Upgrade testing" (omitting mention of Molecule from at least the title text.

Then we can easily link to this exact procedure in its own document to explain the process.